command/cp.go

```diff
@@ -635,16 +679,18 @@ func (c Copy) doUpload(ctx context.Context, srcurl *url.URL, dsturl *url.URL) er
 		}
 	}
 
-	msg := log.InfoMessage{
-		Operation:   c.op,
-		Source:      srcurl,
-		Destination: dsturl,
-		Object: &storage.Object{
-			Size:         size,
-			StorageClass: c.storageClass,
-		},
+	if !c.showProgress {
+		msg := log.InfoMessage{
+			Operation:   c.op,
+			Source:      srcurl,
+			Destination: dsturl,
+			Object: &storage.Object{
+				Size:         size,
```

Comment:
`size` variable is only used here. Could you move the `srcClient.Stat` call inside this if block?

Reply:
`cp` is used by the `mv` command, and `mv` deletes the source object after finishing the upload process. Due to that, we have to call `Stat` before deleting the source object.

Reply:
Thanks for the clarification.